pkg/pipeline/encode/metrics_common.go

```diff
@@ -228,8 +228,8 @@ func (m *MetricsCommonStruct) extractGenericValue(flow config.GenericMap, info *
 	}
 	val, found := flow[info.ValueKey]
 	if !found {
-		m.errorsCounter.WithLabelValues("RecordKeyMissing", info.Name, info.ValueKey).Inc()
-		return nil
+		// No value means 0 to keep storage lightweight
+		return 0
```

Comment:
we should still return nil, I think: for counters, returning 0 or nil will have the same effect (ie. adding nothing to the counter) but for histograms, return 0 would actually corrupt the stats computation

Reply:
It depends of the histograms I guess... Do you have some concrete examples in mind ?

For data consistency we should probably add filters before doing such assumptions. For example, DNS metrics must have DNSId, DNSLatencyMs fields etc. That's what we do on plugin query side.

Reply:
All the histograms that we use are latency stuff, so I don't see a case where we would want to say: no latency == 0ms. If a flow is missing RTT, or Dns latency, then we should just ignore it for histogram stats. Expecting an explicit filter would add extra processing and could easily be error-prone IMO. I'd return your question: do you have an example in mind where we need to say nil == 0 for an histogram ?

Reply:
bytes, packets and their respective drops fields must be considered as 0 when keys are not present

Reply:
yes but that's fine these are counters not histograms, adding 0 to the counter is equivalent to ignoring it